docs: add instructions for using nginx/apache as a reverse proxy for authentication

djach7 opened the pull request:
Creates REVERSE_PROXY.md with instructions for setting up nginx/Apache as a reverse proxy. Also updates SECURITY.md with relevant information. 

Created with help from Claude.

miabbott commented on REVERSE_PROXY.md:
````diff
@@ -0,0 +1,375 @@
+sudo systemctl enable httpd
+```
+
+### 2. Create Password File
````
I believe the passwords will be hashed after they are entered, so I think retrieving them after the fact would be difficult.  Maybe a note here about recording the password for later use would be smart.

miabbott commented on REVERSE_PROXY.md:
```diff
@@ -0,0 +1,375 @@
+# Test configuration
+sudo nginx -t
+
+# Reload if test passes
```
Why is the reload needed?

miabbott commented on REVERSE_PROXY.md:
```diff
@@ -0,0 +1,375 @@
+
+## Nginx Configuration
+
+### Configuration File
```
I trust that all the configuration options here are needed, but I would suggest adding some "flavor text" at the top of this section (or as comments in the config file) about why the options are needed and what they are doing.

djach7 replied:
Addressed in next update

djach7 commented on REVERSE_PROXY.md:
```diff
@@ -0,0 +1,375 @@
+
+## Apache Configuration
+
+### Configuration File
```
Addressed in next update

djach7 commented:
@miabbott Just pushed an update addressing your review, thanks for taking a look at it! I think I addressed everything but please lmk if I missed something

djach7 commented on REVERSE_PROXY.md:
```diff
@@ -0,0 +1,422 @@
+    ssl_certificate_key /path/to/your/key.pem;
+    ssl_protocols TLSv1.2 TLSv1.3;
+    ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384;
+    ssl_prefer_server_ciphers off;
```
Claude disagrees here, stating the following reasons. I will do my own research as well.

  1. TLS 1.3: Only includes secure cipher suites by design, so client preference is safe
  2. Curated cipher list: We've already restricted to strong ECDHE ciphers only
  3. Current standards: Mozilla's "Modern" SSL configuration recommends off for TLS 1.2+1.3 setups

  The on setting was more important in the past when clients might choose weak ciphers, but with our
  restricted cipher suite and TLS 1.3 support, client preference can actually provide better performance while
   maintaining security.

  Reference: https://ssl-config.mozilla.org/

djach7 commented on REVERSE_PROXY.md:
```diff
@@ -0,0 +1,422 @@
+    SSLCertificateKeyFile /path/to/your/key.pem
+    SSLProtocol all -SSLv3 -TLSv1 -TLSv1.1
+    SSLCipherSuite ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384
+    SSLHonorCipherOrder off
```
Claude again disagrees, for these reasons: 

  1. Consistency: Matches our nginx configuration approach
  2. Modern practice: With TLS 1.3 and our curated strong cipher list, client preference is safe
  3. Performance: Can provide better performance with modern clients

  The SSLHonorCipherOrder on setting was more critical in the past when weak ciphers were commonly available.
  Since we've restricted to only strong ECDHE cipher suites and support TLS 1.3, either setting maintains
  security, but off follows current best practices.

  This keeps the Apache and nginx configurations aligned in their security approach.

djach7 commented on REVERSE_PROXY.md:
```diff
@@ -0,0 +1,422 @@
+    SSLCertificateKeyFile /path/to/your/key.pem
+    SSLProtocol all -SSLv3 -TLSv1 -TLSv1.1
+    SSLCipherSuite ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384
+    SSLHonorCipherOrder off
```
Claude again disagrees, for these reasons:

Consistency: Matches our nginx configuration approach
Modern practice: With TLS 1.3 and our curated strong cipher list, client preference is safe
Performance: Can provide better performance with modern clients
The SSLHonorCipherOrder on setting was more critical in the past when weak ciphers were commonly available.
Since we've restricted to only strong ECDHE cipher suites and support TLS 1.3, either setting maintains
security, but off follows current best practices.

This keeps the Apache and nginx configurations aligned in their security approach.

djach7 commented:
/gemini review

djach7 commented:
/gemini review

djach7 commented on REVERSE_PROXY.md:
```diff
@@ -0,0 +1,418 @@
+    # Security Headers
+    Header always set X-Frame-Options DENY
+    Header always set X-Content-Type-Options nosniff
+    Header always set Strict-Transport-Security "max-age=31536000; includeSubDomains"
```
This is intentional, as it makes it easier for users to copy/paste what example applies to them.

kgiusti commented on REVERSE_PROXY.md:
```diff
@@ -0,0 +1,418 @@
+    }
+    
+    # Block everything else
+    location / {
```
Should we provide unauthenticated access to the /fdo/101 API subtree as well?  IIUC FDO doesn't require auth on these endpoints (doing its own auth in the protocol).  Might make sense since the server ports (8038, etc) are firewalled off.

Opinions?

say-paul commented:
LGTM
